addons/web/static/src/js/chrome/loading.js

```diff
@@ -41,36 +48,58 @@ var Loading = Widget.extend({
         this.on_rpc_event(-1);
     },
     on_rpc_event : function(increment) {
-        var self = this;
         if (!this.count && increment === 1) {
             // Block UI after 3s
-            this.long_running_timer = setTimeout(function () {
-                self.blocked_ui = true;
-                framework.blockUI();
-            }, 3000);
+            this.long_running_timer = setTimeout(
+                this.proxy("trigger_up"),
```

Comment:
What is `proxy`? Can you point where is declared?

Reply:
https://github.com/odoo/odoo/blob/3b049e3395ebb53605b54cc352088ddecc753c9d/addons/web/static/src/js/core/mixins.js#L252-L280

Reply:
We have found it, but please change it to current conventions:

```suggestion
                this.trigger_up.bind(this),
```

Reply:
>       *

thanks, I wouldn't find it :)
It's a bit of magic that we don't use anymore

Reply:
Are you sure? That convention should be wrong as you can read in the docstring above.

Reply:
Well, they are pretty sure that they don't use that since at least 2 years

Reply:
Can you think of anything that would change the value of trigger_up between the time the setTimeout starts and the time setTimout actually executes the function ?

Reply:
If you don't use the proxy, then the method that's gonna be called is not gonna be extensible. You can use other alternatives if you prefer, but hard-binding is really not a good idea.

Modern JS equivalent would be: `() => this.trigger_up`

Reply:
I'm good with () => this.trigger_up('loading.blocked')